### DISCUSSION TITLE
Device ESP32 si cui flashare tasmota facilmente

### DISCUSSION THREAD

**Discussion by asmaran85** [indomus/forum#2024] [Hardware & Firmware]
Ciao a tutti.
Ho acquistato su AliExpress la serratura domotica m530 airbnk, una delle poche serrature domotiche che sfrutta serratura preesistente e con un costo basso.
Ho preso la versione zigbee sperando di poterla integrare con Home Assistant.
Purtroppo dopo il pairing non si vede alcuna entità, stessa situazione collegando la serratura al gateway zigbee tuya e quest'ultimo a Home Assistant.
Cercando in rete sembra che l'unico modo per collegare questa serratura ad home assistant sia questa: https://github.com/rospogrigio/airbnk_mqtt 
ed ecco che arriviamo a problema dell'oggetto di questo topic: quale device ESP32 mi consigliate su cui sia super-facile installare tasmota?
Esiste un dispositivo su cui si possa flashare il dispositivo via OTA, senza saldature o altri hardware?
Da quello che ho capito i dispositivi ESP32 non sono tantissimi https://templates.blakadder.com/esp32.html , per il sonoff dual r3 serve fare delle saldature mentre per gli Shelly plus serve un hardware aggiuntivo, esistono alternative?
Grazie

**Comment by snakuzzo** [comment]
Guarda...
premesso che non voglio entrare nel merito della serratura che hai acquistato, né sul perché non riesci ad integrarla su HA (che sia ZHA, Zigbee2mqtt, Zigbee2Tasmota, SLS o Deconz). Non voglio nemmeno entrare nel dettaglio di quel progetto che hai trovato, ma se ti serve un ESP32 non è necessario comprare un sonoff o uno shelly.
Puoi acquistare una board con un semplice esp32. Lo trovi ovunque, anche su Amazon (ti basta cercare "esp32" :) ). Costa pochi euro e lo puoi flashare direttamente via USB.

**Reply by asmaran85** [reply]
Grazie del consiglio. Avevo visto quei dispositivi ma temevo che il firmware tasmota non fosse universale ma adattato per alcuni device.
Proverò come mi consigli.

Purtroppo molti dei device per Tuya usano zigbee per comunicare con il loro gateway ma non sono utilizzabili con ZHA. Zigbee2qmtt ha maggiore compatibilità ma comunque non copre tutti i dispositivi (spero che con il tempo la compatibilità aumenti).
L'hardware in se é molto interessante, il lato software abbastanza penoso.

**Reply by snakuzzo** [reply]
Il firmware Tasmota è flashabile su qualsiasi device che abbia chip ESP826x o ESP32...non ne esistono quelli flashabili o non flashabili...lo sono tutti. Magari può essere più o meno complesso a seconda dei casi. Ci sono quelli su qui tocca saldare e altri dove (come nel tuo caso) basta collegare un cavetto micro USB.

Quanto al discorso Zigbee, sia per ZHA che per Zigbee2mqtt si può contribuire per far supportare nuovi device. Non bisogna per forza stare lì ad aspettare e sperare. Per ZHA magari può essere più complesso (non ho visto nel dettaglio), ma per Zigbee2mqtt c'è una procedura documentata sulla doc di progetto per permettere a "chiunque" di aggiungere nuovi device...ed il risultato lo puoi vedere ogni volta che esce una nuova versione. Sono gli utenti stessi a contribuire, non solo i contributor del progetto.

Leggi [QUI](https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html)
